### PR TITLE
Add Docker HEALTHCHECK to all services using the `grpc_health_probe` executable

### LINKED ISSUE #89: Save a copy of `grpc_health_probe` binary to the repository
# Feature Request

Save a copy of [`grpc_health_probe`](https://github.com/grpc-ecosystem/grpc-health-probe/releases/tag/v0.4.11) binary to the repository. 10MB.

**Is your feature request related to a problem?**

Each of our services downloads this health probe (and will soon use with [Dockerfile HEALTHCHECK](https://docs.docker.com/engine/reference/builder/#healthcheck))) using wget.
Example: https://github.com/open-telemetry/opentelemetry-demo-webstore/blob/main/src/adservice/Dockerfile#L35

**Describe the solution you'd like:**

Instead of downloading each time which a version has to be specified in each Dockerfile or possibly one main `GRPC_HEALTH_PROBE_VERSION` value in the `.env` file that we'd have to add to each service's environment section in the `compose.yml` file, we simply copy the file from `/bin/` to the docker image.

Pros:
- Don't need to install `wget` in each service's build
- No external dependencies (network)
- Don't have to download 10MB file each new build (around 10 times), some issues have been seen with this command
- Simplifies the task of using the same version of the health probe
- Less boilerplate code

Cons:
- Checking in a binary is usually considered a no-no
- Git repository size increases (from 20MB to 30MB)

**Describe alternatives you've considered.**

If the Docker compose command is able to download this binary and it be copied to each image during the build process could alleviate most of the network download required from 100MB total to 10MB total. However, each clean install will still need network access to get this file, so the offline-friendly approach of including the binary to the repository seems worthwhile.

### PR DESCRIPTION
**Pull request by mic-max**
Fixes #89

## Changes

- Includes the gRPC health checking binary to the repository
- Each service will now copy and use this binary as part of the [Docker HEALTHCHECK](https://docs.docker.com/engine/reference/builder/#healthcheck)
- Adds instructions to inspect health of services correctly showing the change works and shows healthy and unhealthy services
 
To be done

- Test and add healtchecking endpoint to all services (not all services currently implement this or not correctly)

### CODE REVIEW

**Comment by julianocosta89** [comment]
The PR looks fine, I'd just check out the discussion going on at #89 to decide if the `grpc_health_probe` will be commited in the repo or downloaded in the Dockerfile.

File: src/paymentservice/README.md [hunk 1]
```diff
@@ -1,3 +1,40 @@
...
+Get the name of the `paymentservice` container using the command
+
+```shell
+docker compose ps
+```
+
+Check the health of the service using a similar command to the following
+
+```shell
+docker inspect --format "{{json .State.Health }}" opentelemetry-demo-webstore-paymentservice-1
```
Comment (mic-max) [inline src/paymentservice/README.md:12]:
```suggestion
docker inspect --format "{{json .State.Health }}" paymentservice
```

File: README.md [hunk 2]
```diff
@@ -86,6 +86,17 @@ After updating the `otelcol-config.yml` start the demo by running
...
+### Check Service Health
+
+Check the health of the service the following command
+
+```shell
+docker compose ps
+```
+
+The status column will indicate whether the service is healthy with `running (health)`
+, unhealthy with `running (unhealthy)`, or simply `running` if health checks are not enabled.
```
Comment (cijothomas) [inline README.md:98]:
```suggestion
The status column will indicate whether the service is healthy with `running (healthy)`
, unhealthy with `running (unhealthy)`, or simply `running` if health checks are not enabled.
```

**Comment by mic-max** [comment]
Converting to draft while I rework this to not include the binary and instead have instructions for users to download the probe to the correct folder and also add a step to the CIs to download the executable.

**Pull request closed without merging**